community/bolt/src/main/java/org/neo4j/bolt/v1/transport/ChunkedOutput.java

```diff
@@ -42,10 +43,12 @@
 
     private final int bufferSize;
     private final int maxChunkSize;
+    private final Semaphore bufferToken = new Semaphore( 1 );
```

Comment:
I don't think there is much point in having a semaphore of size 1. I think using synchronized is sufficient and then we don't have to worry about not releasing the locks